DropboxLike.Api/Controllers/FileController.cs

```diff
@@ -9,43 +9,43 @@ namespace DropboxLike.Api.Controllers;
 public class Filecontroller : ControllerBase
 {
   private readonly ApplicationDbContext _dbContext;
-  private readonly IFileRepository _repository;
+  private readonly FileRepository _repository;
 
-  public Filecontroller(ApplicationDbContext dbContext, IFileRepository repository)
+  public Filecontroller(ApplicationDbContext dbContext, FileRepository repository)
   {
     _dbContext = dbContext;
     _repository = repository;
   }
 
   [HttpPost]
-  public async Task<List<File>> PostAsync(IFormFile files)
+  public async Task<ActionResult<File>> PostAsync(IFormFile file)
   {
-    //Getting FileName
-    var fileName = Path.GetFileName(files.FileName);
-
-    //Getting file Extension
-    var fileExtension = Path.GetExtension(fileName);
-
-    // concatenating  FileName + FileExtension
-    var newFileName = String.Concat(Convert.ToString(Guid.NewGuid()), fileExtension);
-
-    var objFiles = new File()
+    if (file.Length > 0)
     {
-      Id = 0,
-      Name = newFileName,
-      FileType = fileExtension,
-      CreatedOn = DateTime.Now
-    };
-
-    using (var target = new MemoryStream())
-    {
-      files.CopyTo(target);
-      objFiles.DataFiles = target.ToArray();
+      //Getting FileName
+      var fileName = Path.GetFileName(file.FileName);
+
+      //Getting file Extension
+      var fileExtension = Path.GetExtension(fileName);
+
+      // concatenating  FileName + FileExtension
+      var newFileName = String.Concat(Convert.ToString(Guid.NewGuid()), fileExtension);
+
+      var objFiles = new File()
+      {
+        Id = 0,
```

Comment:
I would imagine you can leave this field empty as the ID should be auto-generated by whatever ORM you are using. Since I have not seen the data-layer implementation yet, this is purely speculative, but with EntityFramework it works like this.